spec/models/filters/criteria/filter_for_sub_population_spec.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require_relative 'shared_filter_criteria_context'
+
+RSpec.describe Filters::Criteria::FilterForSubPopulation do
+  include_context 'filter criteria setup'
+
+  let(:sub_population) { :veterans }
+  let(:filter) do
+    ::Filters::FilterBase.new(
+      user_id: user.id,
+      sub_population: sub_population,
+    )
+  end
+  let(:criteria) { described_class.new(input: filter, config: config) }
+
+  # Make sure we have the right scope methods available
+  before do
+    # We need to ensure our scope responds to all the subpopulation methods
+    available_sub_populations = [
+      :adults_with_children, :adults_with_children_youth_hoh,
+      :adults_with_children_twentyfive_plus_hoh, :adult_only_households, :clients,
+      :child_only_households, :non_veterans, :veterans
+    ]
+
+    available_sub_populations.each do |method|
+      allow(scope).to receive(method).and_return(scope)
+    end
+
+    # Set up specific return values for our test cases
+    allow(scope).to receive(:veterans).and_return(scope.where(id: enrollments[0].id))
+    allow(scope).to receive(:non_veterans).and_return(scope.where(id: enrollments[1].id))
+    allow(scope).to receive(:adult_only_households).and_return(scope.where(id: enrollments[2].id))
+  end
```

Comment:
I'm probably missing something, but by specifying which enrollment to return doesn't this bypass calling the actual scope?

This feels more like a test of the test code than the described class.

Maybe this could be something like:
```
available_sub_populations = AvailableSubPopulations.available_sub_populations
```

And then we could loop over those calling `criteria.apply(scope)` and checking against the expected set of enrollments?  Maybe the enrollments need to be grouped into sets keyed by each population?

Reply:
Good point, I've improved it to cover all subpopulations